extensions/notebook/src/book/bookTocManager.ts

```diff
@@ -74,21 +76,58 @@ export class BookTocManager implements IBookTocManager {
 		return toc;
 	}
 
-	updateToc(tableOfContents: JupyterBookSection[], findSection: BookTreeItem, addSection: JupyterBookSection): JupyterBookSection[] {
-		for (const section of tableOfContents) {
-			if ((section as IJupyterBookSectionV1).url && path.dirname(section.url) === path.join(path.sep, path.dirname(findSection.uri)) || (section as IJupyterBookSectionV2).file && path.dirname((section as IJupyterBookSectionV2).file) === path.join(path.sep, path.dirname(findSection.uri))) {
-				if (tableOfContents[tableOfContents.length - 1].sections) {
-					tableOfContents[tableOfContents.length - 1].sections.push(addSection);
-				} else {
-					tableOfContents[tableOfContents.length - 1].sections = [addSection];
-				}
-				break;
+	/**
+	 * Reads and modifies the table of contents file of the target book.
+	 * @param version the version of the target book
+	 * @param tocPath Path to the table of contents
+	 * @param findSection The section that will be modified.
+	 * @param addSection The section that'll be added to the target section. If it's undefined then the target section (findSection) is removed from the table of contents.
+	*/
+	async updateTOC(version: string, tocPath: string, findSection: JupyterBookSection, addSection?: JupyterBookSection): Promise<void> {
```

Comment:
All these version parameters/properties should really be using the enum you created - that just helps with readability and understanding of what the possible values should be